osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -35,11 +36,22 @@ public BeatmapDatabase(BasicStorage storage, BasicGameHost importHost = null)
 
             if (connection == null)
             {
-                connection = storage.GetDatabase(@"beatmaps");
-                connection.CreateTable<BeatmapMetadata>();
-                connection.CreateTable<BaseDifficulty>();
-                connection.CreateTable<BeatmapSetInfo>();
-                connection.CreateTable<BeatmapInfo>();
+                retry:
+                try
+                {
+                    connection = storage.GetDatabase(@"beatmaps");
+                    connection.CreateTable<BeatmapMetadata>();
+                    connection.CreateTable<BaseDifficulty>();
+                    connection.CreateTable<BeatmapSetInfo>();
+                    connection.CreateTable<BeatmapInfo>();
+                }
+                catch
+                {
+                    Debug.WriteLine(@"Beatmap database was unable to be migrated; starting fresh!");
+                    connection?.Close();
+                    storage.DeleteDatabase(@"beatmaps");
+                    goto retry;

```

Comment:
I'd rather avoid retry and instead make it a loop
`do ... while (connection == null);`
Also better add a maximum amount of retries, and rethrow instead of setting connection to null if it is hit. I don't trust SQLite to not end up giving us weirdness aside from when the database is outdated. Having an endless loop instead of an exception to work with would be bad in that case.